### PR TITLE
python3Packages.swspotify: init at 1.2.1, swaglyrics: init at 1.2.2

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/requests/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/requests/default.nix:11]: this is on staging, and the cause of your mass rebuilds. I would drop the commit if you can.

If you can't then I would wait until the next staging cycle gets merged

Comment (siraben) [reply pkgs/development/python-modules/requests/default.nix:11]: I see. Would it be possible to patch the pypi packages somehow to make them depend on requests==2.23.0 instead?

File: pkgs/development/python-modules/swspotify/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/swspotify/default.nix:18]: if you want to follow semver, which requests does
```suggestion
  postPatch = ''
    substituteInPlace setup.py \
      --replace 'requests>=2.24.0' 'requests~=2.23'
  '';
```

File: pkgs/development/python-modules/requests/default.nix
Comment (siraben) [reply pkgs/development/python-modules/requests/default.nix]: Ok, commit dropped.

File: pkgs/development/python-modules/swspotify/default.nix
Comment (siraben) [reply pkgs/development/python-modules/swspotify/default.nix]: Ok, done.

**Comment by siraben** [comment]
@GrahamcOfBorg build swaglyrics

File: pkgs/development/python-modules/swspotify/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/swspotify/default.nix:18]: the difference is important, `~=2.23.0` is essentially `>=2.23,2.23.*` where as `~=2.23` is essentially `>=2.23,2.*`
```suggestion
      --replace 'requests>=2.24.0' 'requests~=2.23'
```

File: pkgs/development/python-modules/swspotify/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/swspotify/default.nix:4]: this commit should be:
```
python3Packages.swspotify: init at 1.2.1
```

File: pkgs/development/python-modules/swspotify/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/swspotify/default.nix:14]: when in `python-modules` do not inherit from python3, as this will always be `python38` and you will have interpreter version issues.

instead, import each python package individually
```suggestion
  propagatedBuildInputs = [
    requests flask-cors dbus-python
  ];
```

File: pkgs/development/python-modules/swspotify/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/swspotify/default.nix:23]: usually description is a small statement
```suggestion
    description = "Library to get the currently playing song and artist from Spotify";
```

File: pkgs/tools/misc/swaglyrics/default.nix
Comment (jonringer) [inline pkgs/tools/misc/swaglyrics/default.nix:18]:
```suggestion
      --replace 'requests>=2.24.0' 'requests~=2.23'
```

File: pkgs/development/python-modules/swspotify/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/swspotify/default.nix:1]: the individual packages also need to be passed as inputs.
```suggestion
{ stdenv, lib, buildPythonPackage, fetchPypi, reqests, flask-cors, dbus-python }:
```

**Comment by siraben** [comment]
Ready to merge.

File: pkgs/development/python-modules/swspotify/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/swspotify/default.nix:20]: There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

```

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

File: pkgs/tools/misc/swaglyrics/default.nix
Comment (jonringer) [inline pkgs/tools/misc/swaglyrics/default.nix:23]: There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

```

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

Comment (siraben) [reply pkgs/tools/misc/swaglyrics/default.nix:23]: There are tests in https://github.com/SwagLyrics/SwSpotify/tree/master/tests , how would I run them?

Comment (jonringer) [reply pkgs/tools/misc/swaglyrics/default.nix]: usually i look at their CI for indications https://github.com/SwagLyrics/SwSpotify/blob/master/.travis.yml

File: pkgs/tools/misc/swaglyrics/default.nix
Comment (jonringer) [inline pkgs/tools/misc/swaglyrics/default.nix:23]: to automatically execute it
```suggestion
  checkInputs = [ python3.pkgs.pytestCheckHook ];
```

File: pkgs/development/python-modules/swspotify/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/swspotify/default.nix:21]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

**Comment by siraben** [comment]
@GrahamcOfBorg build swaglyrics

**Comment by siraben** [comment]
@GrahamcOfBorg build swspotify

**Comment by siraben** [comment]
@GrahamcOfBorg python3Packages.swspotify

**Comment by siraben** [comment]
Currently stale. I suggest merging without checks for now.

**Comment by siraben** [comment]
Ok, rebased on latest master. Enabling tests still causes the build to fail, though.

**Comment by jonringer** [comment]
if they still have unit tests that run within the sandbox, they should be enabled as much as possible. If all the tests aren't able to run, then there should be a comment stating that they aren't able to be ran within a sandbox.

**Comment by siraben** [comment]
I think I realize what's happening now by testing with `nix-shell` after `unpackPhase`, there doesn't seem to be a `tests` folder, is this due to the tests not being uploaded to PyPi?

**Comment by siraben** [comment]
@jonringer I've updated the derivation to disable tests that require network access, need to do the same with the `python3Packages.swspotify` package.

File: pkgs/tools/misc/swaglyrics/default.nix
Comment (jonringer) [inline pkgs/tools/misc/swaglyrics/default.nix:31]: since you're using pytestCheckHook, this should work
```suggestion
  # disable tests which touch network
  disabledTests = [
     "test_database_for_unsupported_song"
     "test_that_lyrics_works_for_unsupported_songs"
     "test_that_get_lyrics_works"
  '';
```

File: pkgs/tools/misc/swaglyrics/skip_network_test.patch
Comment (jonringer) [inline pkgs/tools/misc/swaglyrics/skip_network_test.patch:9]: This patch should be unnecessary with my other comment

File: pkgs/tools/misc/swaglyrics/default.nix
Comment (jonringer) [inline pkgs/tools/misc/swaglyrics/default.nix:30]:
```suggestion
  disabledTests = [
     "test_database_for_unsupported_song"
     "test_that_lyrics_works_for_unsupported_songs"
     "test_that_get_lyrics_works"
  ];
```

File: pkgs/top-level/python-packages.nix
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:3628]: since the time this PR was opened, python-packages was sorted, please attempt to place this in the correct location.

File: pkgs/development/python-modules/swspotify/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/swspotify/default.nix:5]:
```suggestion
  version = "1.2.1";
  disabled = isPy27;
```
```
builder for '/nix/store/4fyyxy5zj2k065yxwqml2vs3l6c1l98w-python2.7-SwSpotify-1.2.1.drv' failed with exit code 4; last 10 log lines:
  tests/test_spotify.py:10: in <module>
      from SwSpotify.spotify import song, artist, get_info_windows, get_info_web
  E     File "/build/source/SwSpotify/spotify.py", line 71
  E       raise SpotifyClosed from None
  E                              ^
  E   SyntaxError: invalid syntax
```

File: pkgs/development/python-modules/swspotify/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/swspotify/default.nix:1]:
```suggestion
{ stdenv, lib, buildPythonPackage, fetchFromGitHub, requests, flask-cors, dbus-python, pytestCheckHook, mock, isPy27 }:
```

File: pkgs/tools/misc/swaglyrics/default.nix
Comment (jonringer) [inline pkgs/tools/misc/swaglyrics/default.nix:29]:
```suggestion
     "test_that_get_lyrics_works"
     "test_lyrics_are_shown_in_tab"
```
```
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='genius.com', port=443): Max retries exceeded with url: /Taylor-Swift-Blank-Space-lyrics (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7ffff5a37850>: Failed to establish a new connection: [Errno -2] Name or service not known'))
```

File: pkgs/tools/misc/swaglyrics/default.nix
Comment (jonringer) [inline pkgs/tools/misc/swaglyrics/default.nix:30]: not sure why this didn't fail in the other review, but it does do a get
```suggestion
     "test_lyrics_are_shown_in_tab"
     "test_songchanged_can_raise_songplaying"
```

**Comment by siraben** [comment]
Just tested on NixOS, tests seem to pass and package runs as expected.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 96051 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary>  <ul>    <li>python37Packages.swspotify</li>    <li>python38Packages.swspotify</li>    <li>swaglyrics</li>  </ul></details>

**Pull request merged**
